classes/ActionScheduler_AdminView.php

```diff
@@ -112,6 +113,94 @@ protected function get_list_table() {
 		return $this->list_table;
 	}
 
+	/**
+	 * Action: admin_notices
+	 *
+	 * Maybe check past-due actions, and print notice.
+	 *
+	 * @uses $this->check_pastdue_actions()
+	 */
+	public function maybe_check_pastdue_actions() {
+
+		# Filter to prevent checking actions (ex: inappropriate user).
+		if ( ! apply_filters( 'action_scheduler_check_pastdue_actions', true ) )
+			return;
+
+		# Get last check transient.
+		$last_check = get_transient( 'action_scheduler_last_pastdue_actions_check' );
+
+		# If transient exists, we're within interval, so bail.
+		if ( !empty( $last_check ) )
+			return;
+
+		# Perform the check.
+		$this->check_pastdue_actions();
+	}
+
+	/**
+	 * Check past-due actions, and print notice.
+	 *
+	 * @todo update $link_url to "Past-due" filter when released (see issue #510, PR #511)
+	 */
+	protected function check_pastdue_actions() {
+
+		# Set thresholds.
+		$threshold_seconds = ( int ) apply_filters( 'action_scheduler_pastdue_actions_seconds', DAY_IN_SECONDS );
+		$threshhold_min    = ( int ) apply_filters( 'action_scheduler_pastdue_actions_min', 1 );
+
+		# Allow third-parties to preempt the default check logic.
+		$check = apply_filters( 'action_scheduler_pastdue_actions_check_pre', null );
+
+		# Scheduled actions query arguments.
+		$query_args = array(
+			'date'     => as_get_datetime_object( time() - $threshold_seconds ),
+			'status'   => ActionScheduler_Store::STATUS_PENDING,
+			'per_page' => $threshhold_min,
+		);
+
+		# If no third-party preempted, run default check.
+		if ( is_null( $check ) ) {
+			$store = ActionScheduler_Store::instance();
+			$num_pastdue_actions = ( int ) $store->query_actions( $query_args, 'count' );
+
+			# Check if past-due actions count is greater than or equal to threshold.
+			$check = ( $num_pastdue_actions >= $threshhold_min );
+			$check = ( bool ) apply_filters( 'action_scheduler_pastdue_actions_check', $check, $num_pastdue_actions, $threshold_seconds, $threshhold_min );
+		}
+
+		# If check failed, set transient and abort.
+		if ( ! boolval( $check ) ) {
+			$interval = apply_filters( 'action_scheduler_pastdue_actions_check_interval', round( $threshold_seconds / 4 ), $threshold_seconds );
+			set_transient( 'action_scheduler_last_pastdue_actions_check', time(), $interval );
+
+			return;
+		}
+
+		# Allow third-parties to conditionally display notice(s).
+		if ( ! apply_filters( 'action_scheduler_show_pastdue_actions_notice', true ) ) {
```

Comment:
@crstauf This filter will work great. It can be moved to the beginning of the function as there is no need to check for past actions if the notice isn't going to be shown.

Reply:
@rrennick I thought about that, but then considered that perhaps a third-party may want `action_scheduler_pastdue_actions_check_pre` to run, and it be logged elsewhere.

Thoughts?

Reply:
See comment above.